### PR TITLE
Fix isolated changes

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

### CODE REVIEW

**Comment by ldclakmal** [comment]
@kalaiyarasiganeshalingam Don't we need to make the `get`, `put` and `invalidate` functions as `isolated` functions?

**Comment by kalaiyarasiganeshalingam** [comment]
> @kalaiyarasiganeshalingam Don't we need to make the `get`, `put` and `invalidate` functions as `isolated` functions?
Yes, we can't bcoz the [remove function ](https://github.com/ballerina-platform/module-ballerina-cache/blob/master/cache-ballerina/src/cache/linked_list.bal#L81) uses a [removeInProgres](https://github.com/ballerina-platform/module-ballerina-cache/blob/master/cache-ballerina/src/cache/linked_list.bal#L85) global variable  and the other two functions call this function.

File: cache-ballerina/src/cache/lru_eviction_policy.bal [hunk 1]
```diff
@@ -50,23 +50,23 @@ public class LruEvictionPolicy {
     # + list - Linked list data structure, which is used to govern the eviction policy
     # + newNode - Node of the linked list, which will be replacing the `oldNode`
     # + oldNode - Node of the linked list, which will be replaced by the `newNode`
-    public function replace(LinkedList list, Node newNode, Node oldNode) {
+    public isolated gitfunction replace(LinkedList list, Node newNode, Node oldNode) {
```
Comment (ldclakmal) [inline cache-ballerina/src/cache/lru_eviction_policy.bal:53]:
```suggestion
    public isolated function replace(LinkedList list, Node newNode, Node oldNode) {
```

File: cache-ballerina/src/cache/linked_list.bal [hunk 2]
```diff
@@ -99,15 +91,14 @@ public function remove(LinkedList list, Node node) {
         }
         node.next = ();
         node.prev = ();
-        removeInProgress = false;
     }
```
Comment (ldclakmal) [inline cache-ballerina/src/cache/linked_list.bal:94]:
```suggestion
```

File: cache-ballerina/src/cache/linked_list.bal [hunk 3]
```diff
@@ -70,20 +70,12 @@ public function addFirst(LinkedList list, Node node) {
...
 # + list - Linked list from which the provided node should be removed
 # + node - The node, which should be removed from the provided linked list
-public function remove(LinkedList list, Node node) {
-    // Using this flag, we prevent the concurrency issues, but this will avoid removing some nodes from the linked-list.
-    // Due to that, when the eviction happens, there can be situations where a node which is used recently is get
-    // removed from the cache.
-    if (!removeInProgress) {
-        removeInProgress = true;
+public isolated function remove(LinkedList list, Node node) {
+
```
Comment (ldclakmal) [inline cache-ballerina/src/cache/linked_list.bal:78]:
```suggestion
```

**Review by ldclakmal — APPROVED** [review]

**Pull request merged**